operator/cmd/mesh/manifest_shared_test.go

```diff
@@ -41,8 +100,226 @@ var (
 	liveReleaseDir string
 	// Path to the operator install base dir in the live release.
 	liveInstallPackageDir string
+
+	// Only used if kubebuilder is installed.
+	testenv               *envtest.Environment
+	testClient            client.Client
+	testReconcileOperator *istiocontrolplane.ReconcileIstioOperator
+
+	allNamespacedGVKs = []schema.GroupVersionKind{
+		{Group: "autoscaling", Version: "v2beta1", Kind: "HorizontalPodAutoscaler"},
+		{Group: "policy", Version: "v1beta1", Kind: "PodDisruptionBudget"},
+		{Group: "apps", Version: "v1", Kind: "Deployment"},
+		{Group: "apps", Version: "v1", Kind: "DaemonSet"},
+		{Group: "", Version: "v1", Kind: "Service"},
+		{Group: "", Version: "v1", Kind: "ConfigMap"},
+		{Group: "", Version: "v1", Kind: "PersistentVolumeClaim"},
+		{Group: "", Version: "v1", Kind: "Pod"},
+		{Group: "", Version: "v1", Kind: "Secret"},
+		{Group: "", Version: "v1", Kind: "ServiceAccount"},
+		{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: "RoleBinding"},
+		{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: "Role"},
+		{Group: "networking.istio.io", Version: "v1alpha3", Kind: "DestinationRule"},
+		{Group: "networking.istio.io", Version: "v1alpha3", Kind: "EnvoyFilter"},
+		{Group: "networking.istio.io", Version: "v1alpha3", Kind: "Gateway"},
+		{Group: "networking.istio.io", Version: "v1alpha3", Kind: "VirtualService"},
+		{Group: "security.istio.io", Version: "v1beta1", Kind: "PeerAuthentication"},
+	}
+
+	// ordered by which types should be deleted, first to last
+	allClusterGVKs = []schema.GroupVersionKind{
+		{Group: "admissionregistration.k8s.io", Version: "v1beta1", Kind: "MutatingWebhookConfiguration"},
+		{Group: "admissionregistration.k8s.io", Version: "v1beta1", Kind: "ValidatingWebhookConfiguration"},
+		{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: "ClusterRole"},
+		{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: "ClusterRoleBinding"},
+		{Group: "apiextensions.k8s.io", Version: "v1beta1", Kind: "CustomResourceDefinition"},
+	}
 )
 
+// TestMain is required to create a local release package in /tmp from manifests and operator/data in the format that
+// istioctl expects. It also brings up and tears down the kubebuilder test environment if it is installed.
+func TestMain(m *testing.M) {
+	var err error
+
+	// If kubebuilder is installed, use that test env for apply and controller testing.
```

Comment:
what happens if it isn't? it will use the real kubeconfig?

Reply:
it would then only run the generate tests, which don't have a kubeconfig. but it would be fairly easy to pass in a real kubeconfig if that's what we want.